scipy/optimize/_lap.py

```diff
@@ -0,0 +1,111 @@
+# Wrapper for the Jonker-Volgenant algorithm for solving the rectangular
+# linear sum assignment problem.  The original code was an implementation of
+# the Hungarian algorithm (Kuhn-Munkres) taken from scikit-learn, based on
+# original code by Brian Clapper and adapted to NumPy by Gael Varoquaux.
+# Further improvements by Ben Root, Vlad Niculae, Lars Buitinck,
+# and Peter Larsen.
+#
+# Copyright (c) 2008 Brian M. Clapper <bmc@clapper.org>, Gael Varoquaux
+# Author: Brian M. Clapper, Gael Varoquaux
+# License: 3-clause BSD
+
+import numpy as np
+from . import _jonkervolgenant
+
+
+def linear_sum_assignment(cost_matrix, maximize=False):
+    """Solve the linear sum assignment problem.
+
+    The linear sum assignment problem is also known as minimum weight matching
+    in bipartite graphs. A problem instance is described by a matrix C, where
+    each C[i,j] is the cost of matching vertex i of the first partite set
+    (a "worker") and vertex j of the second set (a "job"). The goal is to find
+    a complete assignment of workers to jobs of minimal cost.
+
+    Formally, let X be a boolean matrix where :math:`X[i,j] = 1` iff row i is
+    assigned to column j. Then the optimal assignment has cost
+
+    .. math::
+        \\min \\sum_i \\sum_j C_{i,j} X_{i,j}
+
+    where, in the case where the matrix X is square, each row is assigned to
+    exactly one column, and each column to exactly one row.
+
+    This function can also solve a generalization of the classic assignment
+    problem where the cost matrix is rectangular. If it has more rows than
+    columns, then not every row needs to be assigned to a column, and vice
+    versa.
+
+    The method used is the Jonker-Volgenant algorithm.
+
+    Parameters
+    ----------
+    cost_matrix : array
+        The cost matrix of the bipartite graph.
+
+    maximize : bool (default: False)
+        Calculates a maximum weight matching if true.
+
+    Returns
+    -------
+    row_ind, col_ind : array
+        An array of row indices and one of corresponding column indices giving
+        the optimal assignment. The cost of the assignment can be computed
+        as ``cost_matrix[row_ind, col_ind].sum()``. The row indices will be
+        sorted; in the case of a square cost matrix they will be equal to
+        ``numpy.arange(cost_matrix.shape[0])``.
+
+    Notes
+    -----
+    .. versionadded:: 0.17.0
+
+    References
+    ----------
+
+    1. https://en.wikipedia.org/wiki/Assignment_problem
+
+    2. R. Jonker and A. Volgenant. A Shortest Augmenting Path Algorithm for
```

Comment:
It's not clear to me if the attribution of the approach to Jonker--Volgenant is as precise as we may wish: They themselves note that shortest path approaches date back to Tomizawa (1979, doi 10.1002/net.3230010206), and in *Assignment Problems*, by Rainer Burkard, Mauro Dell'Amico, Silvano Martello, the algorithm implemented here is simply referred to as "Shortest_Path" (see Algorithm 4.9, Section 4.4). Most of the work done in Jonker--Volgenant is related to preprocessing and initialization -- coming up with a good partial solution before moving to shortest path searches -- but that hasn't been included in the implementation at hand (and I'm not saying I think it should be). So, is there anything in Jonker--Volgenant's approach that mean that we shouldn't just attribute the approach to any of the earlier references?

Reply:
I have got removed references to all papers but Crouse's.  It is Crouse's pseudocode I have used to implement this module.  Interested readers can look through his references for further info.  The wikipedia page also lists plenty of reading material.